client/app/account/user/agreements/user-agreements.controller.js

```diff
@@ -1,37 +1,30 @@
-angular.module("UserAccount").controller("UserAccountAgreementsController", [
-    "$scope",
-    "$translate",
-    "UserAccount.services.agreements",
-    "Alerter",
-    function ($scope, $translate, Service, Alerter) {
-        "use strict";
-
-        function init () {
-            $scope.loading = true;
-            $scope.list = [];
-
-            $scope.getToValidate();
-        }
-
+angular
+    .module("UserAccount")
+    .controller("UserAccountAgreementsController", ($scope, $translate, Alerter, OvhApiMe, UserAccountAgreements) => {
         $scope.loaders = {
             toActivate: true,
             toActivateList: true
         };
-
         $scope.toActivate = [];
-
         $scope.agreed = {};
 
+        function init () {
+            $scope.loading = true;
+            $scope.list = [];
+
+            return $scope.getToValidate();
+        }
+
         $scope.loadAgreementsList = function (count, offset) {
             init();
 
-            Service.getList(count, offset)
+            return UserAccountAgreements
+                .getList(count, offset)
                 .then((agreements) => {
                     $scope.list = agreements;
-                }, (err) => {
-                    Alerter.error(`${$translate.instant("user_agreements_error")} ${_.get(err, "message") || err}`, "agreements_alerter");
                 })
-                .then(() => {
+                .catch((err) => Alerter.error(`${$translate.instant("user_agreements_error")} ${_.get(err, "message") || err}`, "agreements_alerter"))
```

Comment:
No need to reject promise ?

Reply:
The rejection is done on the `UserAccountAgreements` service. From here, I just want to display relevant message to the user.